### PR TITLE
Introduce JSON response in controller

### LINKED ISSUE #51: Show the original translation when the value is deleted.
https://github.com/user-attachments/assets/e58c00ed-6f39-4230-909d-967df6ab86a5

### PR DESCRIPTION
**Pull request by CuddlyBunion341**
Closes #51 

Added JSON responses to translation_files_controller where necessary. Updated stimulus controller to fetch and display fallback translation to improve UX.

### CODE REVIEW

File: app/controllers/moirai/translation_files_controller.rb
Comment (coorasse) [inline app/controllers/moirai/translation_files_controller.rb:1]: I don't think we should support json format if it's not used. It's code that we don't need. For the actions where we only use HTML, there's no need to support json format.

**Review by coorasse — CHANGES_REQUESTED** [review]
I don't think we need to support json for all the controller actions. Also this makes it difficult to judge to the change that you actually wanted to do. Can you maybe isolate the change in its own PR?

File: app/controllers/moirai/translation_files_controller.rb [hunk 1]
```diff
@@ -115,5 +140,14 @@ def translation_same_as_current?
 
       translation_params[:value] == @file_handler.parse_file(file_paths.first)[translation_params[:key]]
     end
+
+    def get_fallback_translation
```
Comment (CuddlyBunion341) [inline app/controllers/moirai/translation_files_controller.rb:144]: Duplicate code 😓

File: app/controllers/moirai/translation_files_controller.rb [hunk 2]
```diff
@@ -60,6 +70,15 @@ def handle_create
...
       end
 
+      if translation_params[:value].blank? && request.format.json?
+        respond_to do |format|
+          format.json do
+            render json: {fallback_translation: get_fallback_translation}
+          end
+        end
+        return
+      end
```
Comment (CuddlyBunion341) [inline app/controllers/moirai/translation_files_controller.rb:80]: No need for respond_to if the format is already being checked.
```suggestion
      if translation_params[:value].blank? && request.format.json?
        return render json: {fallback_translation: get_fallback_translation}
      end
```

File: test/controllers/moirai/translation_files_controller_test.rb [hunk 3]
```diff
@@ -74,6 +80,12 @@ class TranslationFilesControllerTest < ActionDispatch::IntegrationTest
     assert_equal count_before - 1, Moirai::Translation.count
   end
 
+  test "update translation with blank value json" do
```
Comment (CuddlyBunion341) [inline test/controllers/moirai/translation_files_controller_test.rb:83]:
```suggestion
  test "update translation with blank value JSON" do
```

**Review by coorasse — APPROVED** [review]
Nice! I updated one line and merged!

**Pull request merged**